src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,278 @@
+
+Changes to these definitions can cause type errors in existing code, even if the changes are purely additive.
+Many projects augment built-in type definitions to fill in gaps or make them more strict per the project's needs.
+Modifications to augmented built-in types can introduce type errors for [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) that previously was permitted.
+
+Example: [TypeScript 4.9 improved the accuracy of the built-in `Promise.resolve` types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#better-types-for-promiseresolve).
+That improved type can break existing code relying on `Promise.resolve(...)` calls previously typed as `any` or `unknown` (&lt;4.9) instead of a `Promise<...>` (>=4.9).
+
+## Strict Versioning Options
+
+Given how "breaking" many common type checking changes are, it follows that if TypeScript followed a strict interpretation of semver, _every release would be a major_.
+TypeScript's version strategy options therefore are:
+
+- Allowing changes to the type system in non-major versions _(the strategy in use today)_
+- Only releasing those changes in major versions
+
+Unfortunately, only releasing type system changes in major versions is impractical regardless of how often those major versions are released.
+
+### Frequent Majors
+
+In theory, TypeScript could release a new major version every three or so months instead of a new minor as it does today.
+This would more directly follow traditional semver by indicating type system changes as breaking.
+
+However, doing so would negate much of the benefits of semver for consumers.
+With a major version released every 3 months, consumers would have to update to that major version very frequently.
+The distinction between major and minor versions would lose meaning.
+
+That loss of distinction would be detrimental to consumers because TypeScript does occasionally need to release an API-level breaking change.
+If every release were to be branded as major, then neither developers nor tooling would be able to easily identify the "true" major breakages.
+
+### Infrequent Majors
+
+On the other hand, TypeScript could release major versions infrequently - say, once every six months, or even every year.
+This would minimize the frequency consumers would need to update major versions and experience breakages.
+
+But restricting all "breaking" changes to an occasional new major version would mean bottling up all changes for months at a time.
+New infrequent major versions would contain a significant amount of type system changes all at once.
+Onboarding to new infrequent major versions would be much more difficult than today's minor versions.
+
+Even worse, because the new changes would have so little public user testing in stable versions versions before release, they'd likely contain ma ny more bugs.
```

Comment:
lol, maybe this was a joke, but "ma ny" => "many"

Reply:
heh nope